src/main/java/ca/corefacility/bioinformatics/irida/security/permissions/project/ModifyProjectPermission.java

```diff
@@ -74,18 +68,12 @@ public boolean customPermissionAllowed(Authentication authentication, Project p)
 		// if we've made it this far, then that means that the user isn't
 		// directly added to the project, so check if the user is in any groups
 		// added to the project.
-		final Collection<UserGroupProjectJoin> groups = ugpjRepository.findGroupsByProject(p);
-		for (final UserGroupProjectJoin group : groups) {
+		final Collection<UserGroupProjectJoin> ugpjCollection = ugpjRepository.findGroupsForProjectAndUser(p, u);
```

Comment:
Can you refactor the direct project permission check above to use `getProjectJoinForUser` and if the join exists check that the role is owner instead of looping through all users in the project.

Reply:
Fixed.